diff --git a/android/src/main/java/com/reactnativemenu/MenuView.kt b/android/src/main/java/com/reactnativemenu/MenuView.kt
index 17ed7c6491b60e87fe829fb3f8841424617b4cfb..0c55452172975d695ba73d0d28d0f3add42e91af 100644
--- a/android/src/main/java/com/reactnativemenu/MenuView.kt
+++ b/android/src/main/java/com/reactnativemenu/MenuView.kt
@@ -47,11 +47,13 @@ class MenuView(private val mContext: ReactContext) : ReactViewGroup(mContext) {
     prepareMenu()
   }
 
-  override fun setHitSlopRect(rect: Rect?) {
-    super.setHitSlopRect(rect)
-    mHitSlopRect = rect
-    updateTouchDelegate()
-  }
+  override var hitSlopRect: Rect?
+    get() = super.hitSlopRect
+    set(value) {
+      super.hitSlopRect = value
+      mHitSlopRect = value
+      updateTouchDelegate()
+    }
 
   override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
     return true
diff --git a/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt b/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt
index 4731e1a1528a2de495cc2ae3156c813b64ec6832..e4d2743bd6a038b769c3cf34c68297a1fc44d109 100644
--- a/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt
+++ b/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt
@@ -123,9 +123,9 @@ abstract class MenuViewManagerBase : ReactClippingViewManager<MenuView>() {
   fun setHitSlop(view: ReactViewGroup, @Nullable hitSlop: ReadableMap?) {
     if (hitSlop == null) {
       // We should keep using setters as `Val cannot be reassigned`
-      view.setHitSlopRect(null)
+      view.hitSlopRect = null
     } else {
-      view.setHitSlopRect(
+      view.hitSlopRect = (
               Rect(
                       if (hitSlop.hasKey("left"))
                               PixelUtil.toPixelFromDIP(hitSlop.getDouble("left")).toInt()
@@ -206,7 +206,7 @@ abstract class MenuViewManagerBase : ReactClippingViewManager<MenuView>() {
 
   @ReactProp(name = ViewProps.OVERFLOW)
   fun setOverflow(view: ReactViewGroup, overflow: String?) {
-    view.setOverflow(overflow)
+    view.overflow = overflow
   }
 
   @ReactProp(name = "backfaceVisibility")
